pkg/service/service.go

```diff
@@ -128,14 +128,27 @@ type Service struct {
 
 // NewService creates a new instance of the service handler.
 func NewService(monitorNotify monitorNotify) *Service {
-	return &Service{
-		svcByHash:       map[string]*svcInfo{},
-		svcByID:         map[lb.ID]*svcInfo{},
-		backendRefCount: counter.StringCounter{},
-		backendByHash:   map[string]*lb.Backend{},
-		monitorNotify:   monitorNotify,
-		healthServer:    healthserver.New(),
-		lbmap:           &lbmap.LBBPFMap{},
+
+	if option.Config.EnableHealthCheckNodePort {
+		return &Service{
+			svcByHash:       map[string]*svcInfo{},
+			svcByID:         map[lb.ID]*svcInfo{},
+			backendRefCount: counter.StringCounter{},
+			backendByHash:   map[string]*lb.Backend{},
+			monitorNotify:   monitorNotify,
+			healthServer:    healthserver.New(),
+			lbmap:           &lbmap.LBBPFMap{},
+		}
+	} else {
+		return &Service{
+			svcByHash:       map[string]*svcInfo{},
+			svcByID:         map[lb.ID]*svcInfo{},
+			backendRefCount: counter.StringCounter{},
+			backendByHash:   map[string]*lb.Backend{},
+			monitorNotify:   monitorNotify,
+			healthServer:    nil,
+			lbmap:           &lbmap.LBBPFMap{},
+		}
 	}
```

Comment:
Instead of duplicating the initialization of the Service struct we can simply have a dedicated variable for the `healthServer` and set it to Service.